File: FastBuy-main/Modulo_Server_Cpp/prova3/Ordine.cpp
#include "Ordine.h"

Ordine::Ordine(string user, int cod_ordine, list<Prodotto> lista_prodotti, float prezzo_finale) {
	this->user = user;
	this->cod_ordine = cod_ordine;
	this->lista_prodotti = lista_prodotti;
	this->prezzo_finale = prezzo_finale;
}

string Ordine::getUser() {
	return user;
}

int Ordine::getCodOrdine() {
	return cod_ordine;
}

list<Prodotto> Ordine::getListaProdotti() {
	return lista_prodotti;
}

float Ordine::getPrezzoFinale() {
	return prezzo_finale;
}

File: FastBuy-main/Modulo_Client_CSharp/Modulo_C_sharp/Ordine.cs
using System.Collections.Generic;

namespace Modulo_C_sharp
{
    public class Ordine
    {
        protected string user;
        protected int cod_ordine;
        protected float prezzo_finale;

        public Ordine() { }

        public Ordine(string user, int cod_ordine, float prezzo_finale)
        {
            User = user;
            Cod_ordine = cod_ordine;
            Prezzo_finale = prezzo_finale;
        }

        public string User { get => user; set => user = value; }
        public int Cod_ordine { get => cod_ordine; set => cod_ordine = value; }
        public float Prezzo_finale { get => prezzo_finale; set => prezzo_finale = value; }
    }

    // Classe derivata da Ordine
    public class OrdineConListaProd : Ordine
    {
        protected List<Prodotto> prodotti_ordinati;

        public OrdineConListaProd(string user, int cod_ordine, List<Prodotto> prodotti_ordinati, float prezzo_finale)
        {
            User = user;
            Cod_ordine = cod_ordine;
            Prodotti_ordinati = prodotti_ordinati;
            Prezzo_finale = prezzo_finale;
        }

        public List<Prodotto> Prodotti_ordinati { get => prodotti_ordinati; set => prodotti_ordinati = value; }
    }
}
